Allow MIQ PG config overrides to be included

fbladilo opened the pull request:
@carbonin @bdunne Please review, I have another PR coming with the corresponding configmaps and corresponding template modifications going to pods. 

- Check for the existance of MIQ PG overrides directory, if available, include it in postgresql.conf
- Defaults taken from https://github.com/ManageIQ/manageiq-appliance/blob/master/COPY/etc/manageiq/postgresql.conf.d/01_miq_overrides.conf
- We use include_dir , so more configs can be stacked, PG will process *.conf in a given dir
- POSTGRESQL_CONFIG_DIR is set via environment variable and can be customized at users convenience
- Can be backported to euwe/CF4.5 builds to allow functionality to existing users, all existing deployments are running almost default settings.
- Resolves #1 

We stack after the default customs from SCL, this is how it looks on config : 
```
bash-4.2$ tail -4 /var/lib/pgsql/data/userdata/postgresql.conf 
# Custom OpenShift configuration:
include '/var/lib/pgsql/openshift-custom-postgresql.conf'
# Begin ManageIQ configuration overrides:
include_dir '/etc/manageiq/postgresql.conf.d'
```

bdunne commented on docker-assets/run-postgresql:
```diff
@@ -12,6 +12,19 @@ function set_miq_role() {
+  if [ -d "${POSTGRESQL_CONFIG_DIR}" ]; then
+    cat >> "$PGDATA/postgresql.conf" <<EOF
+# Begin ManageIQ configuration overrides:
+include_dir '${POSTGRESQL_CONFIG_DIR}'
```
Do we need to check if this is already here to avoid having multiple entries in the config?

fbladilo replied:
@bdunne Yep, working on another commit with slightly different logic for safety.

fbladilo commented:
@bdunne @carbonin A few changes : 

* Added a check for the existence of the include_dir entry on postgresql.conf, skip if found.
* Create the default PG_CONF_DIR directory on image, this is safer and guarantees that even if the configmap is removed/missing/non-existant, PG will attempt to come online instead of failing immediately.
* Changed default location of MIQ PG conf.d directory to /var/lib/pgsql (always writable and safer), we want to avoid permissions and related issues by having a default conf.d on /etc.

bdunne commented on docker-assets/run-postgresql:
```diff
@@ -12,6 +12,24 @@ function set_miq_role() {
+
+  # Check for POSTGRESQL_CONFIG_DIR presence in postgresql.conf, if already there skip.
+  [[ $(grep ${POSTGRESQL_CONFIG_DIR} ${PGDATA}/postgresql.conf) ]] && RET=$? || RET=$?
+  if [[ $RET == "0" ]]; then
```
I think `if [[ $? -eq 0 ]]` would be simpler

fbladilo replied:
I tried to avoid checking straight on $? , too many times things get in the middle by accident. That's why I chose to save status on a VAR. Script is run with set -eu and and very picky, will probably quit immediately on a false test.
